addons/hr_recruitment/data/mail_template_data.xml

```diff
@@ -172,34 +171,32 @@
                 Hello,
                 <br/><br/>
                 We confirm we successfully received your application for the job
-                "<a href="${object.job_id.website_url or ''}" style="color:#9A6C8E;"><strong>${object.job_id.name}</strong></a>" at <strong>${object.company_id.name}</strong>.
+                "<a t-att-href="hasattr(object.job_id, 'website_url') and object.job_id.website_url or ''" style="color:#9A6C8E;"><strong t-out="object.job_id.name or ''">Experienced Developer</strong></a>" at <strong t-out="object.company_id.name or ''">YourCompany</strong>.
                 <br/><br/>
                 We will come back to you shortly.
 
-                % if 'website_url' in object.job_id and object.job_id.website_url:
-                    <div style="padding: 16px 8px 16px 8px;">
-                        <a href="${object.job_id.website_url}"
-                            style="background-color: #875a7b; text-decoration: none; color: #fff; padding: 8px 16px 8px 16px; border-radius: 5px;">Job Description</a>
-                    </div>
-                % endif
+                <div t-if="'website_url' in object.job_id and hasattr(object.job_id, 'website_url') and object.job_id.website_url" style="padding: 16px 8px 16px 8px;">
```

Comment:
```
                <div t-if="'website_url' in object.job_id and hasattr(object.job_id, 'website_url') and object.job_id.website_url" style="padding: 16px 8px 16px 8px;">
```
checking if the attribute exists in two different ways seems redundant.

Also treating records as mappings for the purpose of checking their fields seems significantly cleaner and more reliable than involving `hasattr`.

Reply:
Nice catch, I've changed it.